File: python/example/customer_support_triage.py
import asyncio
import os
from baseai import BaseAI
from baseai.tools import BaseAITool


async def main():
    api_key = os.getenv("BASEAI_API_KEY", "YOUR_API_KEY")
    if api_key == "YOUR_API_KEY":
        raise ValueError("Please set BASEAI_API_KEY environment variable or update the api_key in the script")
    client = BaseAI(api_key=api_key, api_url="https://a2abase.ai/api")

    thread = await client.Thread.create()
    desired_name = "Customer Support Triage"
    agent = await client.Agent.find_by_name(desired_name)
    created = False
    if agent is None:
        agent = await client.Agent.create(
            name=desired_name,
            system_prompt=(
                "Monitor support inbox, label priority, suggest responses, and escalate critical issues."
            ),
            mcp_tools=[BaseAITool.WEB_SEARCH_TOOL],
        )
        created = True

    run = await agent.run("Triage today's new tickets and propose responses.", thread)
    stream = await run.get_stream()
    async for line in stream:
        print(line)

    if created:
        await agent.delete()


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/example/customer_support_triage.ts
/**
 * Example: Customer Support Triage Agent
 * Demonstrates web search tools for monitoring and triaging support tickets.
 */
import { A2ABase } from '../src/a2abase';
import { A2ABaseTool } from '../src/tools';

async function main() {
  const apiKey = process.env.BASEAI_API_KEY;
  if (!apiKey || apiKey === 'YOUR_API_KEY') {
    throw new Error('Please set BASEAI_API_KEY environment variable');
  }

  const client = new A2ABase({
    apiKey,
    apiUrl: 'https://a2abase.ai/api',
  });

  const thread = await client.Thread.create();
  const desiredName = 'Customer Support Triage';
  let agent = await client.Agent.findByName(desiredName);
  let created = false;

  if (!agent) {
    agent = await client.Agent.create({
      name: desiredName,
      systemPrompt: 'Monitor support inbox, label priority, suggest responses, and escalate critical issues.',
      a2abaseTools: [A2ABaseTool.WEB_SEARCH_TOOL],
    });
    created = true;
  }

  const run = await agent.run("Triage today's new tickets and propose responses.", thread);
  const stream = await run.getStream();

  for await (const chunk of stream) {
    process.stdout.write(chunk);
  }

  if (created) {
    await agent.delete();
  }
}

main().catch(console.error);

